source/AI.cpp

```diff
@@ -2573,6 +2630,107 @@ bool AI::DoCloak(Ship &ship, Command &command)
 
 
 
+bool AI::DoRoving(Ship &ship, Command &command, unsigned int rovingShipCount)
+{
+	const System* system = ship.GetSystem();
+	if((rovingShipCount > 9 && system))
+	{
+		if(ship.Fuel() >= ship.JumpFuel() || ship.Fuel() >= ship.JumpDriveFuel())
+		{
+			MoveIndependent(ship, command);
+			return true;
+		}
+		else if (system->IsInhabited(&ship))
+		{
+			for(const StellarObject& planet: system->Objects())
+				if(CanRefuel(ship, &planet))
+				{
+					ship.SetTargetStellar(&planet);
+				}
+			Refuel(ship, command);
+			return true;
```

Comment:
This is doing the same thing as the bored personality. Maybe refactor this into another function?